altapay.php

```diff
@@ -0,0 +1,2353 @@
+                if (!$refundUpdate) {
+                    throw new Exception("The refund could not be updated in database");
+                }
+            } catch (Exception $e) {
+                $response = json_decode($e->getMessage(), true);
+                saveLastErrorMessage($paymentID, $response['responseMsg']);
+
+                echo json_encode(
+                    array(
+                        'status' => $response['responseResult'],
+                        'message' => "Could not refund payment. " . $response['responseMsg']
+                    )
+                );
+                die();
+            }
+
+            echo json_encode(
+                array(
+                    'status' => "success",
+                    'message' => "Payment refunded successfully"
+                )
+            );
+            die();
+        } // RELEASE
+        elseif ($action == 'Release') {
+            try {
+                $api->release($paymentID, $action);
+                updatePaymentStatus($paymentID, "Payment Released");
+            } catch (Exception $e) {
+                $response = json_decode($e->getMessage(), true);
+                saveLastErrorMessage($paymentID, $response['responseMsg']);
+
+                echo json_encode(
+                    array(
+                        'status' => $response['responseResult'],
+                        'message' => "Could not release reservation. " . $response['responseResult'] . ": " . $response['responseMsg']
+                    )
+                );
+                die();
+            }
+            echo json_encode(
+                array(
+                    'status' => "success",
+                    'message' => "Reservation released successfully"
+                )
+            );
+            die();
+        }
+    }
+
+    /* ******************************** */
+
+    /**
+     * Method for generating order lines from order backend
+     * @param $orderLines
+     * @param $orderLineGiftWrap
+     * @param $orderID
+     * @param bool $goodWillRefund
+     * @return array
+     */
+    private function populateOrderLinesFromPost($orderLines, $orderLineGiftWrap = null, $orderID, $goodWillRefund = false)
+    {
+        $i = 0;
+        $priceAfterDiscountRounded = 0;
+        $priceAfterDiscount = 0;
+        $totalQuantity = 0;
+        $compensationAmountPerQuantity = 0;
+        $altapayOrderLines = array();
+        $discountPercentage = 0;
+
+
+        $orderDetail = new Order((int)$orderID);
+        $productDetailObject = new OrderDetail;
+
+        $productDetail = $productDetailObject->getList($orderID);
+
+        $compensationQuantity = 0;
+
+        foreach ($orderLines as $key => $orderedQuantity) {
+            if ($orderedQuantity > 0) {
+                $productDetails = $productDetail[$key];
+                if (!empty($productDetails)) {
```

Comment:
Since we know that `$productDetails` has been initialized on the previous line there is no need for using `empty()`, which could leave to hiding a typo in the code.
```suggestion
                if ($productDetails) {
```